### PR TITLE
Adds the `engine-catalog` tool

### LINKED ISSUE #25814: Engine - Develop `engine-catalog` as a replacement for the `wazuh-engine catalog` CLI command
### Description:
The `engine-catalog` script will replace the existing `wazuh-engine catalog` CLI command, inheriting all its functionalities with enhancements to facilitate operations via API calls. This script will manage the catalog of assets within Wazuh-Engine, supporting operations such as adding, removing, updating, validating assets, and retrieving lists of assets by type and namespace. It must handle asset definitions in both YAML and JSON formats, ensuring compatibility and ease of integration within the new architecture.

### Tasks:
1. **Development of the `engine-catalog` Script:**
   - Implement functionalities to add, remove, update, and validate assets in the catalog.
   - Enable fetching of asset lists filtered by type and namespace.

2. **Support for Multiple Data Formats:**
   - Ensure the script can handle asset data in both YAML and JSON formats seamlessly.

3. **Integration with Wazuh-Engine API:**
   - Develop the script to interact with the Wazuh-Engine API for all operations instead of direct CLI commands.

4. **Documentation:**
   - Document the usage, features, and examples of the `engine-catalog` script in the Wazuh-Engine documentation.
   - Update the API and developer guides to include information on the new script.

### Expected Outcome:
A fully functional `engine-catalog` script that provides a robust interface for managing the Wazuh-Engine catalog, replacing the need for the original CLI command and enhancing the system's flexibility and API-driven interaction.

This implementation aims to ensure a smooth transition from the CLI-based asset management to a more scalable and manageable API-driven approach, aligning with the broader goals of enhancing Wazuh-Engine's configurability and ease of use.

### PR DESCRIPTION
**Pull request by juliancnn**
|Related issue|
|---|
| Closes #25814 |

This PR adds the `engine-catalog` python tool that remplace the `wazuh-engine catalog` cli command.

### CODE REVIEW

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/delete.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # Get the params
+    api_socket: str = args['api_socket']
+    namespace: str = args['namespace']
+    asset: str = args['asset']
+
+    # Create the api request
+    try:
+        resource_handler.delete_catalog_file(api_socket, '', asset, namespace)
+    except Exception as e:
+        print(f'Error deleting asset or collection: {e}')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/delete.py:15]: use sys.exit(f'Error deleting asset or collection: {e}')

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/get.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    outFormat: Format = StringToFormat(args['format'])
+
+    # Create the api request
+    try:
+        result = resource_handler.get_catalog_file(
+            api_socket, '', asset, namespace, outFormat)['data']['content']
+        print(result)
+
+    except Exception as e:
+        print(f'Error getting asset or collection: {e}')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/get.py:20]: ditto

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    namespace: str = args['namespace']
+    type: str = args['asset-type']
+    inFormat: Format = StringToFormat(args['format'])
+    path: str = args['path']
+    recursive: bool = args['recursive']
+    abort_on_error: bool = args.get('abort_on_error', False)
+
+    # Check if the path exists and is a directory
+    if not os.path.exists(path):
+        print(f'Error: Path {path} does not exist.')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py:20]: ditto

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        exit(1)
+
+    if not os.path.isdir(path):
+        print(f'Error: Path {path} is not a directory.')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py:24]: ditto

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import sys
+import os
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/load.py:2]: use pathlib

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/update.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    # Read all content from stdin
+    if not content:
+        content = sys.stdin.read()
+
+    # Create the api request
+    try:
+        resource_handler.update_catalog_file(
+            api_socket, '', name, content, namespace, inFormat)
+    except Exception as e:
+        print(f'Error updating asset: {e}')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/update.py:24]: ditto

File: src/engine/tools/engine-suite/src/engine_catalog/cmds/validate.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    # Read all content from stdin
+    if not content:
+        content = sys.stdin.read()
+
+    # validate the api request
+    try:
+        resource_handler.validate_catalog_file(api_socket, '', name, content, namespace, inFormat)
+    except Exception as e:
+        print(f'Error updating asset: {e}')
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/engine_catalog/cmds/validate.py:23]: ditto

File: src/engine/tools/engine-suite/src/shared/resource_handler.py [hunk 8]
```diff
@@ -19,6 +19,16 @@ class Format(Enum):
     YML = auto()
     TEXT = auto()
 
+def StringToFormat(string: str) -> Format:
+    if string == 'json':
+        return Format.JSON
+    elif string == 'yml' or string == 'yaml':
```
Comment (NahuFigueroa97) [inline src/engine/tools/engine-suite/src/shared/resource_handler.py:25]: use only if statement

**Review by NahuFigueroa97 — COMMENTED** [review]
Good job @juliancnn!  Just a few small things. Remember to also add the documentation for these commands to the README.md

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**